Cubical/Algebra/BooleanRing/Base.agda

```diff
@@ -0,0 +1,246 @@
+{-# OPTIONS --safe #-}
+module Cubical.Algebra.BooleanRing.Base where
+
+open import Cubical.Foundations.Prelude hiding (_∧_;_∨_)
+open import Cubical.Foundations.Structure
+open import Cubical.Data.Empty as ⊥
+open import Cubical.Algebra.Ring
+open import Cubical.Algebra.AbGroup.Base
+open import Cubical.Algebra.CommRing
+open import Cubical.Tactics.CommRingSolver
+
+
+private
+  variable
+   ℓ ℓ' : Level
+
+record IsBooleanRing {B : Type ℓ}
+              (𝟘 𝟙 : B) (_+_ _·_ : B → B → B) (-_ : B → B) : Type ℓ where
+
+  field
+    isCommRing   : IsCommRing 𝟘 𝟙 _+_ _·_ -_
+    ·IsIdempotent : (x : B) → x · x ≡ x
+
+  open IsCommRing isCommRing public
+
+record BooleanStr (A : Type ℓ) : Type (ℓ-suc ℓ) where
+  field
+    𝟘          : A
+    𝟙          : A
+    _+_        : A → A → A
+    _·_        : A → A → A
+    -_         : A → A
+    isBooleanRing : IsBooleanRing 𝟘 𝟙 _+_ _·_ -_
+
+  infix  8 -_
+  infixl 7 _·_
+  infixl 6 _+_
+
+  open IsBooleanRing isBooleanRing public
+
+BooleanRing : ∀ ℓ → Type (ℓ-suc ℓ)
+BooleanRing ℓ = TypeWithStr ℓ BooleanStr
+
+BooleanStrToCommRingStr : { A : Type ℓ } →  BooleanStr A  → CommRingStr A
+BooleanStrToCommRingStr x = record { isCommRing = IsBooleanRing.isCommRing (BooleanStr.isBooleanRing x) }
+
+BooleanRingToCommRing : BooleanRing ℓ → CommRing ℓ
+BooleanRingToCommRing (carrier , structure ) = carrier , BooleanStrToCommRingStr structure
+
+module BooleanAlgebraStr (A : BooleanRing ℓ)  where
+  open BooleanStr (A . snd)
+  _∨_ : ⟨ A ⟩ → ⟨ A ⟩ → ⟨ A ⟩
+  a ∨ b = (a + b) + (a · b)
+  _∧_ : ⟨ A ⟩ → ⟨ A ⟩ → ⟨ A ⟩
+  a ∧ b = (a · b)
+  ¬_ : ⟨ A ⟩ → ⟨ A ⟩
+  ¬ a = 𝟙 + a
+
+  variable x y z : ⟨ A ⟩
```

Comment:
It would be more in line with other laws of algebraic structures to use explicit arguments...
(Maybe NAMING.md should be changed to make that clear)

Reply:
It varies and I'll leave it as is